File: java-oop/Girl.java
public class Girl extends Person implements Singer {

    public String sing() {
        return "Lalala";
    }


    public Girl(String name, int age) {
        super(name, age);
    }
    

    @Override
    public String getName() {
        return "girl: " + this.name;
    }

   
}

File: go-oop/model/girl.go
package model

type Girl struct {
	Person
}

func (g Girl) Sing() string {
	return "Lalala"
}

// Override method
func (g *Girl) GetName() string {
	return "girl: " + g.name
}

func NewGirl(name string, age int) Girl {
	return Girl{
		Person: Person{
			name: name,
			age:  age,
		},
	}
}
